src/components/PaginationContainer.ts

```diff
@@ -0,0 +1,150 @@
+import * as dojoLang from "dojo/_base/lang";
+import * as dojoConnect from "dojo/_base/connect";
+
+import { Pagination, PaginationProps } from "./Pagination";
+import { ValidateConfigs } from "./ValidateConfigs";
+import {
+    ListView,
+    PaginationContainerProps,
+    PaginationContainerState,
+    WrapperProps,
+    parseStyle
+} from "../utils/ContainerUtils";
+import "../ui/Pagination.css";
+
+export default class PaginationContainer extends Component<PaginationContainerProps, PaginationContainerState> {
+    private navigationHandler: object;
+    private listListViewHeight: number;
+
+    constructor(props: PaginationContainerProps) {
+        super(props);
+
+        this.state = {
+            findingListviewWidget: true,
+            maxPageSize: 0,
+            offSet: 1,
+            showPageButton: true
+        };
+
+        this.updateListView = this.updateListView.bind(this);
+        this.transformListView = this.transformListView.bind(this);
+
+        this.navigationHandler = dojoConnect.connect(
+            props.mxform,
+            "onNavigation",
+            this,
+            dojoLang.hitch(this, this.validateConfigs)
+        );
+    }
+
+    render() {
+        return createElement("div",
+            {
+                className: classNames("widget-pagination", this.props.class),
+                style: parseStyle(this.props.style)
+            },
+            createElement(ValidateConfigs, {
+                ...this.props as WrapperProps,
+                queryNode: this.state.targetNode,
+                targetListview: this.state.targetListView,
+                validate: !this.state.findingListviewWidget
+            }),
+            this.renderPageButton()
+        );
+    }
+
+    componentWillUnmount() {
+        dojoConnect.disconnect(this.navigationHandler);
+    }
+
+    public static setMessageStatus(currentOffSet: number, offSet: number, maxPageSize: number): string {
+        let fromValue = currentOffSet + 1;
+        let toValue = 0;
+        if (maxPageSize === 0) {
+            fromValue = 0;
+        } else if (maxPageSize < offSet || (currentOffSet + offSet) > maxPageSize) {
+            toValue = maxPageSize;
+        } else {
+            toValue = currentOffSet + offSet;
+        }
+
+        return window.mx.ui.translate(
+            "mxui.lib.MxDataSource",
+            "status",
+            [ fromValue, toValue, maxPageSize ]);
+    }
+
+    private renderPageButton(): ReactElement<PaginationProps> | null {
+        if (this.state.validationPassed) {
+            return createElement(Pagination, {
+                maxPageSize: this.state.maxPageSize,
+                offSet: this.state.offSet,
+                onClickAction: this.updateListView,
+                setMessageStatus: PaginationContainer.setMessageStatus,
+                showPageButton: this.state.showPageButton
+            });
+        }
+
+        return null;
+    }
+
+    private validateConfigs() {
```

Comment:
Seems like this should be happening in the `ValidateConfigs` component.

Reply:
done